app/models/vendor.rb

```diff
@@ -6,4 +6,8 @@ class Vendor < ActiveRecord::Base
   has_many :locations, through: :vendor_location_schedules
 
   validates_presence_of :vendor_name
+
+  def self.approve(vendor)
```

Comment:
We tried doing that at the outset but had trouble getting it to work. Could you help us get that one cleaned up?